homeassistant/components/renson/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""The Renson integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from renson_endura_delta.renson import RensonVentilation
+from requests import RequestException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Renson from a config entry."""
+
+    api = RensonVentilation(entry.data[CONF_HOST])
+    coordinator = RensonCoordinator("Renson", hass, api)
+
+    if not await hass.async_add_executor_job(api.connect):
+        raise ConfigEntryNotReady("Cannot connect to Renson device")
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "api": api,
+        "coordinator": coordinator,
+    }
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class RensonCoordinator(DataUpdateCoordinator):
+    """Data update coordinator for Renson."""
+
+    def __init__(
+        self,
+        name: str,
+        hass: HomeAssistant,
+        api: RensonVentilation,
+        update_interval=timedelta(seconds=30),
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name=name,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=update_interval,
+        )
+        self.api = api
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        try:
+            async with async_timeout.timeout(30):
+                return await self.hass.async_add_executor_job(self.api.get_all_data)
+        except RequestException as err:
```

Comment:
This exception is already handled by default by the base coordinator.

https://github.com/home-assistant/core/blob/a6d6807dd0db5631e711cb24de1e52cb7a95d1cc/homeassistant/helpers/update_coordinator.py#L259

We can remove the try... except block.